internal/provider/resource_gitlab_pages_domain.go

```diff
@@ -0,0 +1,220 @@
+package provider
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_pages_domain", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_pages_domain`" + ` resource enables endpoints for connecting custom domain(s) and TLS certificates in GitLab Pages.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/pages_domains.html)`,
+
+		CreateContext: resourceGitlabPagesDomainCreate,
+		ReadContext:   resourceGitlabPagesDomainRead,
+		UpdateContext: resourceGitlabPagesDomainUpdate,
+		DeleteContext: resourceGitlabPagesDomainDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Description: "The custom domain.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"project": {
+				Description: "The ID or full path of the project which the branch is created against.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"auto_ssl_enabled": {
+				Description: "Enables automatic generation of SSL certificates issued by Let’s Encrypt for custom domains.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    false,
+				Default:     nil,
+			},
+			"certificate": {
+				Description: "The certificate in PEM format with intermediates following in most specific to least specific order.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Default:     nil,
+			},
+			"key": {
+				Description: "The certificate key in PEM format.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Default:     nil,
+			},
+			"url": {
+				Description: "The URL for the given domain.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"certificate_data": {
+				Description: "The certificate data.",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Computed:    true,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"expired": {
```

Comment:
What about the `subject` ?

Reply:
https://github.com/xanzy/go-gitlab/blob/master/pages_domains.go#L44 returns only the expired and expiration keys.

Reply:
... which is somewhat misaligned with the upstream API ... 

In case I have some time tomorrow I'll update the GitLab docs and `go-gitlab`. Otherwise, I think we go with what's there :)